src/main/java/com/jozufozu/flywheel/compat/CompatHelper.java

```diff
@@ -0,0 +1,45 @@
+package com.jozufozu.flywheel.compat;
+
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import com.google.common.base.Suppliers;
+
+import net.fabricmc.loader.api.FabricLoader;
+import net.fabricmc.loader.api.ModContainer;
+import net.fabricmc.loader.api.Version;
+import net.fabricmc.loader.api.metadata.version.VersionPredicate;
+
+public class CompatHelper {
+	public static final Supplier<Boolean> IS_SODIUM_LOADED = Suppliers.memoize(() -> FabricLoader.getInstance().isModLoaded("sodium"));
+	public static final Supplier<Boolean> IS_EMBEDDIUM_LOADED = Suppliers.memoize(() -> FabricLoader.getInstance().isModLoaded("embeddium"));
+
+	static {
+		boolean is0_5Loaded = false;
+		boolean is0_6Loaded = false;
+
+		if (IS_SODIUM_LOADED.get() && !IS_EMBEDDIUM_LOADED.get()) {
+			Optional<ModContainer> containerOptional = FabricLoader.getInstance()
+					.getModContainer("sodium");
+
+			if (containerOptional.isPresent()) {
+				Version sodiumVersion = containerOptional.get()
+						.getMetadata()
+						.getVersion();
+
+				try {
+					VersionPredicate predicate0_5 = VersionPredicate.parse(">=0.5.0 <0.6.0");
+					VersionPredicate predicate0_6 = VersionPredicate.parse("<0.6 >=0.6.0-beta.2");
```

Comment:
These ranges are also incorrect. The first one should be `~0.5.0 >=0.5.0` and the second one should be `>=0.6.0-beta.2`. Again, double check specifically in Minecraft that this is correct.